examples/eks_test_fixture/main.tf

```diff
@@ -41,15 +41,24 @@ locals {
 
   worker_groups = [
     {
-      instance_type       = "t2.small"
-      additional_userdata = "echo foo bar"
-      subnets             = "${join(",", module.vpc.private_subnets)}"
+      # This will launch an autoscaling group with only On-Demand instances
+      instance_type        = "t2.small"
+      additional_userdata  = "echo foo bar"
+      subnets              = "${join(",", module.vpc.private_subnets)}"
+      asg_desired_capacity = "2"
     },
+  ]
+  worker_groups_launch_template = [
     {
-      instance_type                 = "t2.small"
-      additional_userdata           = "echo foo bar"
-      subnets                       = "${join(",", module.vpc.private_subnets)}"
-      additional_security_group_ids = "${aws_security_group.worker_group_mgmt_one.id},${aws_security_group.worker_group_mgmt_two.id}"
+      # This will launch an autoscaling group with only Spot Fleet instances
+      instance_type                            = "t2.small"
+      additional_userdata                      = "echo foo bar"
+      subnets                                  = "${join(",", module.vpc.private_subnets)}"
+      additional_security_group_ids            = "${aws_security_group.worker_group_mgmt_one.id},${aws_security_group.worker_group_mgmt_two.id}"
+      override_instance_type                   = "t3.small"
+      asg_desired_capacity                     = "2"
+      spot_instance_pools                      = 10
```

Comment:
Does not really make sense to request 2 instances across 10 spot pools. Such configuration is confusing, especially when put in examples. I would go with something along `asg_desired_capacity = 4` and `spot_instance_pools = 2` - 2x2 instance types = four instances across two spot pools (types).

Reply:
I'm gonna merge this now as it's been open and on-going for a while now. The LT/ASG arrangement is also completely optional for now and not used by default. By merging this people can at least have a play with it for now.

@zytek perhaps you could make a PR with the changes suggested here?

Reply:
Awesome. I'll play with it and can make a PR.
Could you push new version to terraform repository ?

Reply:
> Could you push new version to terraform repository ?

Sure but in a couple days. This repo has sucked up enough of my time for today. You can always point your module source directly at github until a new release is created 🙂

Reply:
FYI:

```
module "eks" {
  source = "git::https://github.com/terraform-aws-modules/terraform-aws-eks.git?ref=whatever_ref_you_want_goes_here"
}